src/tts_text/utils.py

```diff
@@ -182,4 +190,15 @@ def get_soup(url: str, dynamic: bool = False) -> BeautifulSoup:
 
         html = response.text
 
-    return BeautifulSoup(html, "html.parser")
+    soup: BeautifulSoup = BeautifulSoup("")
+    if retries is None:
+        while not soup.contents:
+            soup = BeautifulSoup(html, "html.parser")
+    elif retries > 0:
+        for _ in range(retries):
+            soup = get_soup(url=url, dynamic=True)
+            if soup.contents:
+                break
```

Comment:
Called if none of the retries worked:

```suggestion
                break
        else:
            raise RuntimeError(f"Could not parse the URL {url}.")
```